py3status/modules/wwan.py

```diff
@@ -339,56 +352,61 @@ def _organize(self, data):
 
     def wwan_status_nm(self):
         modem_proxy = self._get_modem_proxy()
-        wwan_data = self._organize(self._get_status(modem_proxy))
+        wwan_data = self._organize(self._get_modem_status_data(modem_proxy))
         network_state = wwan_data.get('state', 0)
         wwan_data['state_name'] = self.network_states[network_state]
 
         # no signal data if less than 8
         if network_state >= 8:
 
-            bearer = self._get_bearer(modem_proxy)
+            name = '_name'
 
             # access technologies and name
             if self.init['access_technologies_name']:
                 key = 'access_technologies'
-                new_key = '%s_name' % key
                 if wwan_data[key]:
                     bit = 1 << (wwan_data[key].bit_length() - 1)
                 else:
                     bit = 0
-                wwan_data[new_key] = self.network_speed[bit]
+                wwan_data[key + name] = self.network_speed[bit]
 
             # modem band
-            if self.init['current_bands']:
+            if self.init['current_bands_name']:
                 key = 'current_bands'
-                new_key = '%s_name' % key
-                wwan_data[new_key] = self.modem_bands[wwan_data[key]]
+                wwan_data[key + name] = self.modem_bands[wwan_data[key]]
 
             # registration state and name
             if self.init['m3gpp_registration_name']:
                 key = 'm3gpp_registration_state'
-                new_key = key.replace('state', 'name')
-                wwan_data[new_key] = self.registration_states[wwan_data[key]]
-
-            # interface name
-            if self.init['interface_name']:
-                wwan_data['interface_name'] = self._get_interface(bearer)
-
-            # ipv4 and ipv6 network config
-            if self.init['ip']:
-                network_config = self._get_network_config(bearer)
-                formats = {'ipv4': self.format_ipv4, 'ipv6': self.format_ipv6}
-                for x in self.init['ip']:
-                    wwan_data['format_' + x] = self.py3.safe_format(
-                        formats[x], network_config.get(x, {}))
-
-            # network connection statistics
-            if self.init['stats']:
-                stats = self._organize(self._get_stats(bearer))
-                stats['duration_hms'] = str(
-                    timedelta(seconds=stats.get('duration', 0)))
-                wwan_data['format_stats'] = self.py3.safe_format(
-                    self.format_stats, stats)
+                wwan_data[key
+                          + name] = self.registration_states[wwan_data[key]]
```

Comment:
Fix this one... Also, `Py3status Telcom` looks better. More real.  And give your `interface_name` eg example too. And have you tested all examples? Some examples are too similar too, I think.

Reply:
I'm sorry i don't understand what you mean with this diff, i'm tired, maybe for this.
Interface example given and more real ip.
I will retest all example tomorrow.

Reply:
The lines shouldn't be broken up (like this) if it went over the column a little. I will review this again, but AFAIK, it looks great and should be ready.